[BUG] Opensearch Dashboards getting 500 internal server error while fetching search aggregation data

**Describe the bug**
Referring to the below commit which is making version specific changes to 2.8.0. Looks like for Dashboards versions above 2.8.0,  this is a breaking change. Currently, we are seeing functional test failures which might be related to this on both windows and linux while running on main (3.0.0) version. We don't see this issue in our 2.x branch (which is currently not bumped to 2.8.1 yet). 

https://github.com/opensearch-project/OpenSearch/pull/7514

**To Reproduce**
Steps to reproduce the behavior:

We can reproduce the error in local with sample flight test data and load the Dashboard page. 

<img width="1765" alt="Screenshot 2023-06-16 at 11 20 08 AM" src="https://github.com/opensearch-project/OpenSearch/assets/105884062/671e9d9a-eeac-422c-b120-dcf69d17e0c4">

```
{
	"statusCode": 500,
	"error": "Internal Server Error",
	"message": "search_phase_execution_exception: [response_handler_failure_transport_exception] Reason: java.lang.NullPointerException: Cannot invoke \"org.opensearch.common.io.stream.DelayableWriteable.asSerialized(org.opensearch.common.io.stream.Writeable$Reader, org.opensearch.common.io.stream.NamedWriteableRegistry)\" because the return value of \"org.opensearch.search.query.QuerySearchResult.aggregations()\" is null",
	"attributes": {
		"error": {
			"root_cause": [{
				"type": "response_handler_failure_transport_exception",
				"reason": "java.lang.NullPointerException: Cannot invoke \"org.opensearch.common.io.stream.DelayableWriteable.asSerialized(org.opensearch.common.io.stream.Writeable$Reader, org.opensearch.common.io.stream.NamedWriteableRegistry)\" because the return value of \"org.opensearch.search.query.QuerySearchResult.aggregations()\" is null"
			}],
			"type": "search_phase_execution_exception",
			"reason": "",
			"phase": "fetch",
			"grouped": true,
			"failed_shards": [{
				"shard": 0,
				"index": "opensearch_dashboards_sample_data_flights",
				"node": "91-QImtGTB2vg2_m2Y07HA",
				"reason": {
					"type": "response_handler_failure_transport_exception",
					"reason": "java.lang.NullPointerException: Cannot invoke \"org.opensearch.common.io.stream.DelayableWriteable.asSerialized(org.opensearch.common.io.stream.Writeable$Reader, org.opensearch.common.io.stream.NamedWriteableRegistry)\" because the return value of \"org.opensearch.search.query.QuerySearchResult.aggregations()\" is null",
					"caused_by": {
						"type": "null_pointer_exception",
						"reason": "Cannot invoke \"org.opensearch.common.io.stream.DelayableWriteable.asSerialized(org.opensearch.common.io.stream.Writeable$Reader, org.opensearch.common.io.stream.NamedWriteableRegistry)\" because the return value of \"org.opensearch.search.query.QuerySearchResult.aggregations()\" is null"
					}
				}
			}],
			"caused_by": {
				"type": "null_pointer_exception",
				"reason": "Cannot invoke \"org.opensearch.search.aggregations.InternalAggregations.getSerializedSize()\" because \"reducePhase.aggregations\" is null"
			}
		}
	}
}
```

**Expected behavior**
Dashboards should be able to consume search pipeline API response without any issue. 

Opensearch Dashboards version   > 2.8.0

**Additional context**
Add any other context about the problem here.

This is the error from OS side
```
[2023-06-21T17:21:20,329][INFO ][o.o.a.s.QueryPhaseResultConsumer] [runTask-0] Created ReducedQueryPhase with totalHits: 0 hits, numReducePhases: 1
[2023-06-21T17:21:20,329][WARN ][r.suppressed             ] [runTask-0] path: /opensearch_dashboards_sample_data_flights/_search, params: {ignore_unavailable=true, preference=1687368079458, index=opensearch_dashboards_sample_data_flights, timeout=30000ms, track_total_hits=true}
org.opensearch.action.search.SearchPhaseExecutionException: 
        at org.opensearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:664) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.opensearch.action.search.FetchSearchPhase$1.onFailure(FetchSearchPhase.java:128) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:54) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.opensearch.threadpool.TaskAwareRunnable.doRun(TaskAwareRunnable.java:78) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.opensearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:59) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:806) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
        at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "org.opensearch.search.aggregations.InternalAggregations.getSerializedSize()" because "reducePhase.aggregations" is null
        at org.opensearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:171) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.opensearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:137) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.opensearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:123) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
        ... 8 more

```

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/60_empty.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/60_empty.yml
@@ -1,0 +1,29 @@
+---
+"Empty Agg Body":
+  - do:
+      index:
+        index: test
+        id: 1
+        body: { "double" : 42 }
+
+  - do:
+      index:
+        index: test
+        id: 2
+        body: { "double" : 100 }
+
+  - do:
+      index:
+        index: test
+        id: 3
+        body: { "double" : 50 }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        body: { "size" : 0, "aggs" : { } }
+
+  - match: { hits.total: 3 }

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/60_empty.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/60_empty.yml
@@ -1,5 +1,5 @@
 ---
-"Empty Agg Body":
+"Empty aggs Body":
   - do:
       index:
         index: test

--- a/server/src/main/java/org/opensearch/search/aggregations/AggregatorFactories.java
+++ b/server/src/main/java/org/opensearch/search/aggregations/AggregatorFactories.java
@@ -235,7 +235,7 @@ public class AggregatorFactories {
             }
         }
 
-        return factories;
+        return factories.count() > 0 ? factories : null;
     }
 
     public static final AggregatorFactories EMPTY = new AggregatorFactories(new AggregatorFactory[0]);

